### PR TITLE
Contribution Guidelines

### PR DESCRIPTION
**Pull request by rrbutani**
Some issue and PR templates, a contribution guide, and a code of conduct.

Since these are organization wide things that affect the entire group, I wanted to have everyone sign off on this PR.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ assignees: ''
 ### steps to reproduce
  - steps
```
Comment (gipsond) [inline .github/ISSUE_TEMPLATE/bug.md]: Consider
```md
- [[step 1]]
- [[...]]
```
or something along those lines instead, for consistent syntax (`[[]]`) with our other comment blocks in the templates.

File: .github/ISSUE_TEMPLATE/improvement.md [hunk 2]
```diff
@@ -30,6 +30,20 @@ this should be:
...
+i.e.: add a `Display` impl for `PriorityLevel`
+
+for such improvements, it's helpful to link to these bits of code here; we ask that you use permalinks to code in `master` to do so
+as an example: [lc3-isa `Word` type](https://github.com/ut-utp/core/blob/4816ece0f2d47e54d989c7a5bc4da9c9f5415f74/isa/src/lib.rs#L82-L83)
+
+see [here](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files) for details about how to make permalinks on GitHub
+
+for improvements where this isn't applicable, feel free to remove this list
+]]
+relevant-code:
```
Comment (gipsond) [inline .github/ISSUE_TEMPLATE/improvement.md]: Recommend another line break here, between `]]` and `relevant-code:`.  They show up on the same line when looking at the rendered text and it's hard to read at first.

File: .github/ISSUE_TEMPLATE/improvement.md [hunk 3]
```diff
@@ -30,6 +30,20 @@ this should be:
...
+  - [](https://github.com/ut-utp/core/blob/4816ece0f2d47e54d989c7a5bc4da9c9f5415f74/isa/Cargo.toml#L2)
```
Comment (gipsond) [inline .github/ISSUE_TEMPLATE/improvement.md]: What is this link?

**Review by gipsond — COMMENTED** [review]
See comments.

Overall comment:
Personally not a huge fan of the all-lowercase, punctuation-optional style, since I feel it's hard to read in some places and maybe a little bit too informal. Happy to keep it that way, though, unless others take greater issue with it.

File: .github/ISSUE_TEMPLATE/improvement.md (hunk 3 above)
Comment (rrbutani) [reply .github/ISSUE_TEMPLATE/improvement.md]: An example; I can mark it as such.

File: .github/ISSUE_TEMPLATE/feature.md [hunk 4]
```diff
@@ -9,24 +9,26 @@ assignees: rrbutani
 
 ### what
 
-[[what is this feature about]]
+[[ What is this feature about. ]]
```
Comment (gipsond) [inline .github/ISSUE_TEMPLATE/feature.md]: `.` should be `?`

**Review by gipsond — DISMISSED** [review]
One minor comment. Otherwise lgtm.

File: .github/ISSUE_TEMPLATE/feature.md (hunk 4 above)
Comment (rrbutani) [reply .github/ISSUE_TEMPLATE/feature.md]: Good catch!

**Review by gipsond — DISMISSED** [review]
lgtm

**Review by jer-zhang — DISMISSED** [review]

**Review by jerryyang98 — DISMISSED** [review]

**Review by pranav12321 — DISMISSED** [review]

**Review by DavidRollins — DISMISSED** [review]

**Pull request closed without merging**